roachtest: knex failed

roachtest.knex [failed](https://teamcity.cockroachdb.com/buildConfiguration/BUILDTYPE_ID-not-found-in-env/5213353?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/BUILDTYPE_ID-not-found-in-env/5213353?buildTab=artifacts#/knex) on master @ [92947c29c55ff909f50a5e625811d34a1bbe71f7](https://github.com/cockroachdb/cockroach/commits/92947c29c55ff909f50a5e625811d34a1bbe71f7):


```
			            	  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/canary.go:138
			            	  | [...repeated from below...]
			            	Wraps: (2) all attempts failed for install knex npm dependencies
			            	Wraps: (3) attached stack trace
			            	  -- stack trace:
			            	  | main.(*clusterImpl).RunE
			            	  | 	main/pkg/cmd/roachtest/cluster.go:1970
			            	  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.repeatRunE
			            	  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/canary.go:131
			            	  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.registerKnex.func1
			            	  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/knex.go:103
			            	  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.registerKnex.func2
			            	  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/knex.go:128
			            	  | main.(*testRunner).runTest.func2
			            	  | 	main/pkg/cmd/roachtest/test_runner.go:876
			            	  | runtime.goexit
			            	  | 	GOROOT/src/runtime/asm_amd64.s:1581
			            	Wraps: (4) output in run_100527.129116164_n1_cd
			            	Wraps: (5) cd /mnt/data1/knex/ && sudo npm i returned
			            	  | stderr:
			            	  | npm WARN EBADENGINE Unsupported engine {
			            	  | npm WARN EBADENGINE   package: 'chai-subset-in-order@3.1.0',
			            	  | npm WARN EBADENGINE   required: { node: '>=v14.16.0' },
			            	  | npm WARN EBADENGINE   current: { node: 'v12.22.12', npm: '8.10.0' }
			            	  | npm WARN EBADENGINE }
			            	  | npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
			            	  | npm WARN deprecated har-validator@5.1.5: this library is no longer supported
			            	  | npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
			            	  | npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
			            	  | npm WARN deprecated tar@2.2.2: This version of tar is no longer supported, and will not receive security updates. Please upgrade asap.
			            	  | npm ERR! code 243
			            	  | npm ERR! path /mnt/data1/knex/node_modules/better-sqlite3
			            	  | npm ERR! command failed
			            	  | npm ERR! command sh -c prebuild-install || npm run build-release
			            	  | npm ERR! prebuild-install WARN install EACCES: permission denied, access '/root/.npm'
			            	  |
			            	  | npm ERR! A complete log of this run can be found in:
			            	  | npm ERR!     /root/.npm/_logs/2022-05-20T10_05_28_040Z-debug-0.log
			            	  |
			            	  | stdout:
			            	Wraps: (6) COMMAND_PROBLEM
			            	Wraps: (7) Node 1. Command with error:
			            	  | ``````
			            	  | cd /mnt/data1/knex/ && sudo npm i
			            	  | ``````
			            	Wraps: (8) exit status 243
			            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.withPrefix (5) *cluster.WithCommandDetails (6) errors.Cmd (7) *hintdetail.withDetail (8) *exec.ExitError
			Test:       	knex

	test_impl.go:265,require.go:1234,knex.go:106,knex.go:128,test_runner.go:876: 
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*knex.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-15093

--- a/pkg/cmd/roachtest/tests/knex.go
+++ b/pkg/cmd/roachtest/tests/knex.go
@@ -65,7 +65,7 @@ func registerKnex(r registry.Registry) {
 			c,
 			node,
 			"add nodesource repository",
-			`sudo apt install ca-certificates && curl -sL https://deb.nodesource.com/setup_12.x | sudo -E bash -`,
+			`sudo apt install ca-certificates && curl -fsSL https://deb.nodesource.com/setup_16.x | sudo -E bash -`,
 		)
 		require.NoError(t, err)
 
@@ -101,7 +101,7 @@ func registerKnex(r registry.Registry) {
 		require.NoError(t, err)
 
 		err = repeatRunE(
-			ctx, t, c, node, "install knex npm dependencies", `cd /mnt/data1/knex/ && sudo npm i`,
+			ctx, t, c, node, "install knex npm dependencies", `cd /mnt/data1/knex/ && npm i`,
 		)
 		require.NoError(t, err)
 

